distributed/scheduler.py

```diff
@@ -3513,8 +3646,9 @@ def workers_to_close(
         if n is None and memory_ratio is None:
             memory_ratio = 2
 
+        ws: WorkerState
         with log_errors():
-            if not n and all(ws.processing for ws in self.workers.values()):
+            if not n and all([ws._processing for ws in self.workers.values()]):
```

Comment:
Ideally this wouldn't need to instantiate the `list`, but ran into some issues with the type being picked up in Cython ( https://github.com/cython/cython/issues/3928 ). So decided to do it this way instead. Expect it doesn't matter much either way since we already track all `Worker`s anyways.

As a side note, we may get more value from the Cython side by converting this to a `for`-loop. That said, will save that for future work should we find this part of the code is slow.

Reply:
Sounds like a sensible plan